src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,146 @@
+            }
+            fw.close();
+        }catch(IOException e){
+            System.out.println("File Saving failed..." + e);
+        }
+    }
+
+    /**
+     * Removes the task at the given index in the list from the stored data.
+     * @param index the given index of the task to be removed.
+     */
+    public void removeListStateRecord(int index) {
+        this.listStates.remove(index);
+        this.writeList();
+    }
+
+    /**
+     * Add task into the stored data using the given type and data.
+     *
+     * @param type the type of the task to be added.
+     *             Can be todo, deadline, or event.
+     * @param data the data of the task to be added in the form of String array,
+     *             with varying size depending on the type.
+     *             For todos, the data only has description.
+     *             For deadlines, the data has description and the deadline.
+     *             For events, the data has description, start time and end time.
+     */
+    public void addListStateRecord(String type, String[] data) {
+        String newRecord = "";
+        switch (type){
+        case "todo":
+            String[] recordT = {"T", "0", data[0]};
+            newRecord = String.join(" | ", recordT);
+            break;
+        case "deadline":
+            String[] recordD = {"D", "0", data[0], data[1]};
+            newRecord = String.join(" | ", recordD);
+            break;
+        case "event":
+            String[] record = {"E", "0", data[0], data[1], data[2]};
+            newRecord = String.join(" | ", record);
+            break;
+        default:
+            break;
+        }
+        this.listStates.add(newRecord);
+        this.writeList();
+    }
+
+    /**
+     * Modify the task at the given index in the stored data,
+     * by changing its mark/unmark state depending on the value of isMarking given
+     *
+     * @param isMarking the value to indicate whether to mark or unmark the taks.
+     *                  true to mark, and false to unmark.
+     * @param index the index of the task to modify.
+     */
+    public void modifyStateRecord(boolean isMarking, int index) {
+        String[] record = this.listStates.get(index).split(" \\| ");
+        record[1] = isMarking ? "1" : "0";
+        String newRecord = String.join(" | ", record);
+        this.listStates.set(index, newRecord);
+        this.writeList();
+    }
+
+    /**
+     * Initializes the directory and file of the stored list.
+     * If the directory does not exist, it is created.
+     * If the file does not exist, it is created.
+     */
+    public void initializeListFile() {
+        try{
+            File dir = new File("./data/");
+            if (!dir.exists()) {
+                if (dir.mkdirs()) {
+//                    System.out.println("Initializaing....");
+                } else {
+//                    System.out.println("Failed to create the directory.");
+                    return;
+                }
```

Comment:
The `if`-block seems a bit redundant.

```suggestion
            if (!dir.exists()) {
                if (!dir.mkdirs()) {
                    return;
                }
```